php/WP_Mock/Hook.php

```diff
@@ -25,6 +25,8 @@ protected function safe_offset( $value ) {
 			return 'null';
 		} elseif ( is_scalar( $value ) ) {
 			return $value;
+		} elseif ( ''.$value === '<Closure>' ) {
+			return '__CLOSURE__';
```

Comment:
hey @acobster may I know why the concatenation with `''.$value` here? shouldn't you check `is_string( $value ) && $value === '<Closure>'` instead? is this expected to have a toString method? or isn't this expected to be `$value instanceof \Closure`? Just double checking, I've read your previous explanation in your PR description.

On the other hand, if it's a string instead, wouldn't it return true for the previous condition `is_scalar()`? 

sorry if I missed anything here

Reply:
The code in `trunk` has [a new case to check for instances of `AnyInstance`](https://github.com/10up/wp_mock/blob/d0584171d602e60d1d58e2e0dccd35dcce529f18/php/WP_Mock/Hook.php#L30-L31) in this method.

Maybe this PR can check for [instances of `Mockery\Matcher\Type`](https://github.com/mockery/mockery/blob/3c3ca93df1752eb33ea9cfa9481c974e650b604c/library/Mockery/Matcher/Type.php#L23), which is what we will receive when `WP_Mock\Functions::type('callable')` is passed as the expected callback?

Reply:
@wvega-godaddy I've updated the code with your feedback in a separate branch `feature/closures-bis` mind checking if that's what you had in mind?